### PR TITLE
[TACHYON-1109] Added checkstyle rule for detecting merge conflicts

### PR DESCRIPTION
**Pull request by yupeng9**
https://tachyon.atlassian.net/browse/TACHYON-1109

### CODE REVIEW

**Comment by apc999** [comment]
Have you tested the checker off line?

**Comment by yupeng9** [comment]
Yep, tested.

File: build/checkstyle/tachyon_checks.xml [hunk 1]
```diff
@@ -31,6 +31,14 @@
   <module name="FileTabCharacter">
     <property name="eachLine" value="true"/>
   </module>
+  <module name="RegexpSingleline">
+    <property name="format" value="&gt;&gt;&gt;&gt;&gt;&gt;&gt;"/>
+    <property name="message" value="Merge conflicts unsolved."/>
```
Comment (gpang) [inline build/checkstyle/tachyon_checks.xml]: "unresolved"?

File: build/checkstyle/tachyon_checks.xml [hunk 2]
```diff
@@ -31,6 +31,14 @@
...
+  </module>
+  <module name="RegexpSingleline">
+    <property name="format" value="&lt;&lt;&lt;&lt;&lt;&lt;&lt;"/>
+    <property name="message" value="Merge conflicts unsolved."/>
```
Comment (gpang) [inline build/checkstyle/tachyon_checks.xml]: "unresolved"?

**Comment by yupeng9** [comment]
Jenkins, test this please

**Comment by calvinjia** [comment]
LGTM

**Comment by apc999** [comment]
@yupeng9  could you resolve merge conflicts? This check is pretty nice to have in checkers

**Comment by haoyuan** [comment]
Jenkins, test this please.

**Comment by apc999** [comment]
+1 LGTM

**Pull request merged**